pkg/brctl/brctl.go

```diff
@@ -0,0 +1,339 @@
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package brctl
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+// Addbr adds a bridge with the provided name.
+func Addbr(name string) error {
+	brctlSocket, err := unix.Socket(unix.AF_INET, unix.SOCK_STREAM, 0)
+	if err != nil {
+		return fmt.Errorf("unix.Socket: %w", err)
+	}
+
+	if _, err := executeIoctlStr(brctlSocket, unix.SIOCBRADDBR, name); err != nil {
+		return fmt.Errorf("executeIoctlStr: %w", err)
+	}
+
+	return nil
+}
+
+// Delbr deletes a bridge with the name provided.
+func Delbr(name string) error {
+	brctlSocket, err := unix.Socket(unix.AF_INET, unix.SOCK_STREAM, 0)
+	if err != nil {
+		return fmt.Errorf("unix.Socket: %w", err)
+	}
+
+	if _, err := executeIoctlStr(brctlSocket, unix.SIOCBRDELBR, name); err != nil {
+		return fmt.Errorf("executeIoctlStr: %w", err)
+	}
+
+	return nil
+}
+
+// Addif adds an interface to the bridge provided
+func Addif(bridge string, iface string) error {
+	brctlSocket, err := unix.Socket(unix.AF_INET, unix.SOCK_STREAM, 0)
+	if err != nil {
+		return fmt.Errorf("unix.Socket: %w", err)
+	}
+
+	ifr, err := unix.NewIfreq(bridge)
+	if err != nil {
+		return fmt.Errorf("unix.NewIfreq: %w", err)
+	}
+
+	ifIndex, err := getIndexFromInterfaceName(iface)
+	if err != nil {
+		return fmt.Errorf("getIndexFromInterfaceName: %w", err)
+	}
+	ifr.SetUint32(uint32(ifIndex))
+
+	if err := unix.IoctlIfreq(brctlSocket, unix.SIOCBRADDIF, ifr); err != nil {
+		return fmt.Errorf("unix.IoctlIfreq: %w", err)
+	}
+
+	return nil
+}
+
+// Delif deleted a given interface from the bridge
+func Delif(bridge string, iface string) error {
+	brctlSocket, err := unix.Socket(unix.AF_INET, unix.SOCK_STREAM, 0)
+	if err != nil {
+		return fmt.Errorf("unix.Socket: %w", err)
+	}
+
+	ifr, err := unix.NewIfreq(bridge)
+	if err != nil {
+		return fmt.Errorf("unix.NewIfreq: %w", err)
+	}
+
+	ifIndex, err := getIndexFromInterfaceName(iface)
+	if err != nil || ifIndex == 0 {
+		return fmt.Errorf("getIndexFromInterfaceName: %w", err)
+	}
+	ifr.SetUint32(uint32(ifIndex))
+
+	if err := unix.IoctlIfreq(brctlSocket, unix.SIOCBRDELIF, ifr); err != nil {
+		return fmt.Errorf("unix.IoctlIfreq: %w", err)
+	}
+
+	return nil
+}
+
+// All bridges are in the virtfs under /sys/class/net/<name>/bridge/<item>, read info from there
+// Update this function if BridgeInfo struct changes
+func getBridgeInfo(name string) (BridgeInfo, error) {
+	basePath := path.Join(BRCTL_SYS_NET, name, "bridge")
+	bridgeID, err := os.ReadFile(path.Join(basePath, BRCTL_BRIDGEID))
+	if err != nil {
+		return BridgeInfo{}, fmt.Errorf("os.ReadFile: %w", err)
+	}
+
+	stpEnabled, err := os.ReadFile(path.Join(basePath, BRCTL_STP_STATE))
+	if err != nil {
+		return BridgeInfo{}, fmt.Errorf("os.ReadFile: %w", err)
+	}
+
+	stpEnabledBool, err := strconv.ParseBool(strings.TrimSuffix(string(stpEnabled), "\n"))
```

Comment:
I think this is the last one! https://google.github.io/styleguide/go/decisions#variable-name-vs-type
```
stpEnabled
```